src/components/spectrogram/spectrogram.ts

```diff
@@ -342,15 +408,49 @@ export class SpectrogramComponent extends SignalWatcher(ChromeHost(LitElement))
       }
 
       if (this.unitConverters.value && change.has("melScale")) {
-        this.unitConverters.value.melScale.value = this.melScale;
+        this.unitConverters.value.melScale.value = this.spectrogramOptions.melScale;
       }
     } else if (this.invalidateSpectrogramSource(change)) {
       this.renderSpectrogram();
     }
   }
 
-  public async renderSpectrogram() {
-    if (!this.hasSource()) {
+  public resetSettings(): void {
+    this.colorMap = "audacity";
+    this.contrast = 1;
+    this.brightness = 0;
+    this.melScale = false;
+
+    this.stop();
+  }
```

Comment:
this seems problematic.

Reset to what? IMO it only makes sense to revert component settings (props) back to the initial state (i.e. reading a copy when component was first made).

Why are we resetting only a subset of them?